src/sentry/issues/auto_source_code_config/task.py

```diff
@@ -0,0 +1,257 @@
+from __future__ import annotations
+
+import logging
+from collections.abc import Mapping
+from enum import StrEnum
+from typing import TYPE_CHECKING, Any
+
+from sentry_sdk import set_tag, set_user
+
+from sentry import eventstore
+from sentry.constants import ObjectStatus
+from sentry.db.models.fields.node import NodeData
+from sentry.integrations.models.repository_project_path_config import RepositoryProjectPathConfig
+from sentry.integrations.services.integration import RpcOrganizationIntegration, integration_service
+from sentry.integrations.source_code_management.metrics import (
+    SCMIntegrationInteractionEvent,
+    SCMIntegrationInteractionType,
+)
+from sentry.issues.auto_source_code_config.code_mapping import CodeMapping, CodeMappingTreesHelper
+from sentry.locks import locks
+from sentry.models.organization import Organization
+from sentry.models.project import Project
+from sentry.models.repository import Repository
+from sentry.shared_integrations.exceptions import ApiError
+from sentry.utils.locking import UnableToAcquireLock
+from sentry.utils.safe import get_path
+
+logger = logging.getLogger(__name__)
+
+if TYPE_CHECKING:
+    from sentry.integrations.base import IntegrationInstallation
+
+
+class DeriveCodeMappingsErrorReason(StrEnum):
+    UNEXPECTED_ERROR = "Unexpected error type while calling `get_trees_for_org()`."
+    LOCK_FAILED = "Failed to acquire lock"
+    EMPTY_TREES = "The trees are empty."
+
+
+def process_event(project_id: int, group_id: int, event_id: str) -> None:
```

Comment:
I have renamed this function from `auto_source_code_configs`.

Now `group_id` is `int` rather than `int | None = None`.